### PR TITLE
Eliminate redundant prompts for acknowledgement if page content has not changed

### PR DESCRIPTION
**Pull request by annda**
With this PR we update the reference revision only when page content has changed.

### CODE REVIEW

File: helper.php [hunk 1]
```diff
@@ -70,16 +70,34 @@ public function removePage($page)
...
+            if ($info['type'] !== DOKU_CHANGE_TYPE_MINOR_EDIT) {
+                // compare content
+                $oldContent = str_replace(NL, '', io_readFile(wikiFN($page, $rev)));
+                $newContent = str_replace(NL, '', $newContent);
+                if ($oldContent !== $newContent) $identical = false;
+                break;
+            }
+        }
+
+        if ($identical) return;
```
Comment (splitbrain) [inline helper.php:99]: if the last 20 revisions were minor, or no previous revisions exist at all, $identical will be true but maybe shouldn't.

File: helper.php [hunk 2]
```diff
@@ -70,16 +70,34 @@ public function removePage($page)
...
-    public function storePageDate($page, $lastmod)
+    public function storePageDate($page, $lastmod, $newContent)
     {
         $sqlite = $this->getDB();
         if (!$sqlite) return;
 
+        $identical = true;
+        $changelog = new \dokuwiki\ChangeLog\PageChangeLog($page);
+
+        $revs = $changelog->getRevisions(1, 20);
```
Comment (splitbrain) [inline helper.php:86]: why 20?

**Review by splitbrain — CHANGES_REQUESTED** [review]
Thinking about this a bit more, would't it be enough to always only compare to the previous revision (regardless if minor or not)?

Please also adjust testing for travis -> github

**Comment by annda** [comment]
- [x] always compare with previous version, this eliminates the need for magic number (20) of revisions to fetch
- [x] Travis -> GitHub

**Pull request merged**